x/migration/types/message_claim_morse_application.go

```diff
@@ -36,59 +36,62 @@ func NewMsgClaimMorseApplication(
 }
 
 // ValidateBasic ensures that:
-// - The morseSignature length is valid (signature validation performed elsewhere).
-// - The morseSrcAddress is valid (i.e. it is a valid hex-encoded address).
 // - The shannonDestAddress is valid (i.e. it is a valid bech32 address).
+// - The application service config is valid.
+// - The morsePublicKey is valid.
+// - The morseSrcAddress matches the public key.
+// - The morseSignature is valid.
 func (msg *MsgClaimMorseApplication) ValidateBasic() error {
-	if len(msg.MorseSignature) != MorseSignatureLengthBytes {
-		return ErrMorseApplicationClaim.Wrapf(
-			"invalid morse signature length; expected %d, got %d",
-			MorseSignatureLengthBytes, len(msg.MorseSignature),
+	// Validate the shannonDestAddress is a valid bech32 address.
+	if _, err := sdk.AccAddressFromBech32(msg.GetShannonDestAddress()); err != nil {
```

Comment:
Should we check if its empty first?

Reply:
This will catch both empty and invalid addresses with reasonable error messages. I think it's a bit redundant to add a check for the empty case.